pkg/alertmanager/multitenant.go

```diff
@@ -478,6 +478,14 @@ func (am *MultitenantAlertmanager) starting(ctx context.Context) (err error) {
 	}
 
 	if am.cfg.ShardingEnabled {
+		// Make sure that all the alertmanagers we were initially configured with have
+		// fetched state from the replicas, before advertising as ACTIVE and letting
+		// them shut-down. This will reduce the possibility that we lose state when
+		// scaling up or down.
+		if err := am.waitInitialStateSync(ctx); err != nil {
```

Comment:
I would add a log before and after otherwise, in case an alertmanager get stuck waiting the initial sync for any reason, may be difficult to spot that was the cause.